examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -1368,6 +1485,113 @@ static bool version_at_least(mongocxx::v_noabi::database& db, int minimum_major)
     return server_major >= minimum_major;
 }
 
+// https://jira.mongodb.org/browse/CXX-2505
+static void queryable_encryption_api(mongocxx::client& client) {
+    // Start Queryable Encryption Example
+    using namespace mongocxx;
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_array;
+    using bsoncxx::builder::basic::make_document;
+    using bsoncxx::types::bson_value::value;
+
+    // Drop data from prior test runs.
+    client["keyvault"]["datakeys"].drop();
+    client["docsExamples"].drop();
+
+    // Create two data keys.
+    class client key_vault_client {
+        uri{}, add_test_server_api(),
+    };
+
+    options::client_encryption ce_opts;
+    ce_opts.key_vault_client(&key_vault_client);
+    ce_opts.key_vault_namespace({"keyvault", "datakeys"});
+    ce_opts.kms_providers(_make_kms_doc(false));
+    client_encryption client_encryption(std::move(ce_opts));
+
+    auto key1_id = client_encryption.create_data_key("local");
+    auto key2_id = client_encryption.create_data_key("local");
+
+    // Create an encryptedFieldsMap.
+    auto encrypted_fields_map = make_document(kvp(
+        "docsExamples.encrypted",
+        make_document(kvp(
+            "fields",
+            make_array(make_document(kvp("path", "encryptedIndexed"),
+                                     kvp("bsonType", "string"),
+                                     kvp("keyId", key1_id),
+                                     kvp("queries", make_document(kvp("queryType", "equality")))),
+                       make_document(kvp("path", "encryptedUnindexed"),
+                                     kvp("bsonType", "string"),
+                                     kvp("keyId", key2_id)))))));
+
+    // Create an FLE 2 collection.
+    options::auto_encryption auto_encrypt_opts{};
+    auto_encrypt_opts.key_vault_namespace({"keyvault", "datakeys"});
+    auto_encrypt_opts.kms_providers(_make_kms_doc(false));
+    auto_encrypt_opts.encrypted_fields_map(encrypted_fields_map.view());
+
+    // Optional, If mongocryptd is not in PATH, then find the binary at MONGOCRYPTD_PATH.
+    char* mongocryptd_path = std::getenv("MONGOCRYPTD_PATH");
+    if (mongocryptd_path) {
+        auto_encrypt_opts.extra_options(
+            make_document(kvp("mongocryptdSpawnPath", mongocryptd_path)));
+    }
+
+    mongocxx::options::client encrypted_client_opts;
+    encrypted_client_opts.auto_encryption_opts(std::move(auto_encrypt_opts));
+    class client encrypted_client {
+        uri{}, add_test_server_api(encrypted_client_opts)
+    };
+
+    // Create the FLE 2 collection docsExample.encrypted.
+    // Because docsExample.encrypted is in encryptedFieldsMap, it is created with FLE 2 support.
```

Comment:
```suggestion
    // Create the Queryable Encryption collection docsExample.encrypted.
    // Because docsExample.encrypted is in encryptedFieldsMap, it is created with Queryable Encryption support.
```